mobile/src/foss/java/org/openhab/habdroid/ui/MapViewHelper.kt

```diff
@@ -0,0 +1,219 @@
+                setBuiltInZoomControls(true)
+                setMultiTouchControls(true)
+                isVerticalMapRepetitionEnabled = false
+                overlays.add(CopyrightOverlay(itemView.context))
+                onResume()
+                handler.post {
+                    applyPositionAndLabel(boundItem, labelView.text, 16.0f,
+                            true, true, this@OsmViewHolder)
+                } }
+            }
+        }
+
+}
+
+fun MapView.applyPositionAndLabel(item: Item?, itemLabel: CharSequence?, zoomLevel: Float,
+                                  allowDrag: Boolean, allowScroll: Boolean,
+                                  markerDragListener: Marker.OnMarkerDragListener) {
+    if (item == null) {
+        return
+    }
+    val canDragMarker = allowDrag && !item.readOnly
+    if (!item.members.isEmpty()) {
+        val positions = ArrayList<GeoPoint>()
+        for (member in item.members) {
+            val position = member.state?.asLocation?.toGeoPoint()
+            if (position != null) {
+                setMarker(position, member, member.label, canDragMarker, markerDragListener)
+                positions.add(position)
+            }
+        }
+
+        if (!positions.isEmpty()) {
+            var north = -90.0
+            var south = 90.0
+            var west = 180.0
+            var east = -180.0
+            for (position in positions) {
+                north = Math.max(position.latitude, north)
+                south = Math.min(position.latitude, south)
+
+                west = Math.min(position.longitude, west)
+                east = Math.max(position.longitude, east)
+            }
+
+            Log.d(MapViewHelper.TAG, String.format("North %f, south %f, west %f, east %f",
+                    north, south, west, east))
+            val boundingBox = BoundingBox(north, east, south, west)
+            val extraPixel = context.resources.dpToPixel(24f).toInt()
+            try {
+                zoomToBoundingBox(boundingBox, false, extraPixel)
+            } catch (e: Exception) {
+                Log.d(MapViewHelper.TAG, "Error applying markers", e)
+            }
+
+            if (!allowScroll) {
+                setScrollableAreaLimitLongitude(west, east, extraPixel)
+                setScrollableAreaLimitLatitude(north, south, extraPixel)
+            }
+        }
+    } else {
+        val position = item.state?.asLocation?.toGeoPoint()
+        if (position != null) {
+            setMarker(position, item, itemLabel, canDragMarker, markerDragListener)
+            controller.setZoom(zoomLevel.toDouble())
+            controller.setCenter(position)
+            if (!allowScroll) {
+                setScrollableAreaLimitLatitude(position.latitude, position.latitude, 0)
+                setScrollableAreaLimitLongitude(position.longitude, position.longitude, 0)
+            }
+        }
+    }
+}
+
+fun MapView.setMarker(pos: GeoPoint, item: Item, label: CharSequence?, canDrag: Boolean,
```

Comment:
`MapView` can be null here. The current master crashes because of this.

Reply:
How? It's initialized in the constructor or created explicitly for the dialog.

Reply:
Nevermind, the crash is related to an internal osm lib bug.